### DISCUSSION TITLE
how to run a schetch on my arduino uno

### DISCUSSION THREAD

**Discussion by dhell1** [community/community#53621] [Programming Help]
### Select Topic Area

Question

### Body

how to run a schetch on my arduino uno

**Comment by jeallz** [comment]
To run a sketch on an Arduino Uno, follow these steps:

> 1. Install the **Arduino IDE** from the official Arduino website.
> 2. Connect your Arduino Uno to your computer using a USB cable.
> 3. Open the sketch (`.ino` file) in the Arduino IDE.
> 4. Go to **Tools → Board** and select **Arduino Uno**.
> 5. Go to **Tools → Port** and choose the COM port where your Arduino is connected.
> 6. Click the **Verify (✓)** button to compile the sketch and check for errors.
> 7. Click the **Upload (→)** button to upload the sketch to the Arduino board.
> 8. Once the upload is complete, the sketch will start running automatically.

If you're having trouble, could you share:

> * Which operating system are you using (Windows, macOS, or Linux)?
> * Are you getting any error messages during compilation or upload?
> * Which version of the Arduino IDE are you using?

**Comment by Mehrab-Javadzadeh** [comment]
Getting started is simple. First, install the Arduino IDE. Then, plug your Arduino Uno into your computer with a USB cable. Open the sketch file you want to use. In the IDE, go to Tools, select your Board as Arduino Uno, and then choose the correct COM port under Tools. Hit the Verify button to check for errors. If it's all good, click Upload.

The sketch will run on the Arduino as soon as it finishes uploading. It'll keep running every time you power up the board, unless you upload something else.

If you get any errors, just tell us what they say, and we can help you figure it out.

**Comment by prashantkoirala465** [comment]
If you mean uploading and running a sketch (Arduino program) on an Arduino Uno, the basic process is:

1. Download and install the Arduino IDE:
https://www.arduino.cc/en/software

2. Connect your Arduino Uno to your computer using a USB cable.

3. Open the Arduino IDE.

4. Select your board:
Tools → Board → Arduino AVR Boards → Arduino Uno

5. Select the correct port:
Tools → Port → choose the port that appears when your Arduino is connected.

6. Open or write your sketch.

For example, you can test it with the built-in LED:

```cpp
void setup() {
  pinMode(LED_BUILTIN, OUTPUT);
}

void loop() {
  digitalWrite(LED_BUILTIN, HIGH);
  delay(1000);
  digitalWrite(LED_BUILTIN, LOW);
  delay(1000);
}
